forge-gui/res/cardsfolder/upcoming/chiss_goria_forge_tyrant.txt

```diff
@@ -0,0 +1,17 @@
+Name:Chiss-Goria, Forge Tyrant
+ManaCost:6 R R
+Types:Legendary Creature Dragon
+PT:5/4
+K:Affinity:Artifact
+K:Flying
+K:Haste
+T:Mode$ Attacks | ValidCard$ Card.Self | TriggerZones$ Battlefield | Execute$ TrigExile | TriggerDescription$ Whenever CARDNAME attacks, exile the top five cards of your library. You may cast an artifact spell from among them this turn. If you do, it has affinity for artifacts.
+SVar:TrigExile:DB$ Dig | Defined$ You | DigNum$ 5 | ChangeNum$ All | DestinationZone$ Exile | RememberChanged$ True | SubAbility$ DBEffect
+SVar:DBEffect:DB$ Effect | StaticAbilities$ STPlay | Triggers$ Play1 | RememberObjects$ Remembered | ForgetOnMoved$ Exile | SubAbility$ DBCleanup
+SVar:STPlay:Mode$ Continuous | EffectZone$ Command | AffectedZone$ Exile,Stack | AddKeyword$ Affinity:Artifact | Affected$ Card.IsRemembered+Artifact | MayPlay$ True | Description$ You may play one of those cards until end of turn.
```

Comment:
play / cast aren't the same
(it doesn't allow _Ancient Den_)

Reply:
I wonder would it be better to use ExiledwithSource here or is this ok ?

Reply:
Ideally yes but for now missing support for timestamp and in ForgetOnMoved make that too unreliable